example/src/app/home.tsx

```diff
@@ -17,13 +22,33 @@ export interface ITodoMutations {
   removeTodo: (input: { id: string }) => void;
 }
 
+const handleSubscription = (type, data, todo) => {
+  switch (type) {
+    case 'todoAdded':
+      if (data.todos.find(t => t.id === todo.id) === undefined) {
+        data.todos.push(todo);
+      }
+      break;
+    case 'todoRemoved':
+      data.todos.splice(data.todos.findIndex(t => t.id === todo.id), 1);
+      break;
+  }
+
+  return data;
+};
+
 const Home: React.SFC<{}> = () => (
   <Connect
     query={createQuery(TodoQuery)}
     mutations={{
       addTodo: createMutation(AddTodo),
       removeTodo: createMutation(RemoveTodo),
     }}
+    subscriptions={[
+      createSubscription(TodoAdded),
+      createSubscription(TodoRemoved),
+    ]}
+    updateSubscription={handleSubscription}
```

Comment:
I see what you're going for here. I think we could simplify this by declaring the subscription handling logic when calling createSubscription.

Something like this maybe?

```tsx
// Connect prop
subscriptions={[
  createSubscription(TodoAdded, handleTodoAdded),

// Outside of component
const handleTodoAdded = (response: TodoAddition, data: any[]) => 
  [...data, { id: response.id, text: response.text }]
```

Reply:
I do like that. I was just mirroring the API from v0 for now. @kitten what do you think?

Reply:
I would regard all these creators as potential functions that might be removed actually if we're planning support for graphql-tag. So depending on them might make things harder.

Also coupling this logic to the creator might not make much sense if this transformation function assumes that there's other queries and mutations that need reconciling. If we assume that these creators might be abstracted away by users (as queries and other things like this are shared) having this transformer on the factory function might give off the impression that it's related to the subscription and not to the other pieces of data

Reply:
Those are amazing points. Let's leave it as is. it'll make the migration from v0-v1 simpler for subscriptions since the API is a near mirror.